lib/dialects/abstract/connection-manager.js

```diff
@@ -271,7 +281,18 @@ class ConnectionManager {
     }
 
     return promise.then(() => {
-      return this.pool.acquire(options.priority, options.type, options.useMaster)
+      return new Promise((resolve, reject) => {
```

Comment:
May be old code but this is no longer required, we now return error to pool and handle that when resolving connection